process/smartContract/processProxy/processProxy.go

```diff
@@ -0,0 +1,182 @@
+package processProxy
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/smartContractResult"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract/processorV2"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract/scrCommon"
+	"github.com/ElrondNetwork/elrond-go/state"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+var log = logger.GetOrCreate("processProxy")
+
+var _ scrCommon.TestSmartContractProcessor = (*scProcessorProxy)(nil)
+
+type configuredProcessor string
+
+const (
```

Comment:
Why using strings? We can use uint8 and still be clean
```go
type configuredProcessor string
const (
  procV1 configuredProcessor = iota,
  procV2
)
...
```

Reply:
fixed